File: algorithms/238_product_of_array_except_self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> result;
        result.push_back(1);
        int med=1;
        for (int i=0;i<nums.size()-1;i++) {
              med=med*nums[i];
              result.push_back(med);}
        med=1;
        for (int i=0;i<nums.size();i++) {
            result[nums.size()-1-i]=result[nums.size()-1-i]*med;
            med=nums[nums.size()-1-i]*med;
        }
        return result;
    }
};

File: algorithms/python3/238_product_of_array_except_self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res=[1]*len(nums)
        res[0]=1
        for i in range(len(nums)-1):
            res[i+1]=res[i]*nums[i]
        cur=1
        for i in range(len(nums)-1,-1,-1):
            res[i]=cur*res[i]
            cur=cur*nums[i]

        return res

a=Solution()

nums=[1,2,3,4]
print(a.productExceptSelf(nums))
